File: OdonSys-SPA/src/app/core/models/api/clients/update-client-request.ts
import { Country } from "../../../../core/enums/country.enum";

export class UpdateClientRequest {
  constructor(
    protected id: string,
    protected name: string,
    protected middleName: string,
    protected surname: string,
    protected secondSurname: string,
    protected phone: string,
    protected country: Country,
    protected email: string,
    protected document: string,
    protected active: boolean
  ) { }
}

File: OdonSysBackEnd/Resource/Contract/Contract.Administration/Clients/UpdateClientRequest.cs
using Utilities.Enums;

namespace Contract.Administration.Clients;

public record UpdateClientRequest
(
    string Id,
    string Name,
    string MiddleName,
    string Surname,
    string SecondSurname,
    string Phone,
    string Email,
    bool Active
)
{
    public Country? Country { get; set; }
    public string Document { get; set; }
};
